core/src/main/java/com/linecorp/armeria/server/HttpHeaderPathMapping.java

```diff
@@ -57,10 +58,10 @@
     HttpHeaderPathMapping(PathMapping pathStringMapping, Set<HttpMethod> supportedMethods,
                           List<MediaType> consumeTypes, List<MediaType> produceTypes) {
         this.pathStringMapping = requireNonNull(pathStringMapping, "pathStringMapping");
-        this.supportedMethods = requireNonNull(supportedMethods, "supportedMethods");
+        this.supportedMethods = ImmutableSet.copyOf(requireNonNull(supportedMethods, "supportedMethods"));
```

Comment:
There has to be a way to make sure that the resulting `Set` is `ImmutableEnumSet`. How about converting `supportedMethods` to `EnumSet` first? e.g.

```java
this.supportedMethods = ImmutableSet.copyOf(EnumSet.copyOf(requireNonNull(supportedMethods, "supportedMethods")));
```

Reply:
Perhaps we could stop sweating about keeping `AbstractRouteBuilder.supportedMethods` is an `EnumSet` or an `ImmutableEnumSet` once we make this change here.